File: tests/vb/vb_conversion_methods/convert_to_int32_from_string.vb
' vybe-test: vb/vb_conversion_methods/convert_to_int32_from_string
' origin: languages/vb/tests/vb/test_vb_conversion_methods.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Imports System

Module M
    Sub Main()
        __Check(CStr(Convert.ToInt32("123")), "123")
        __Check(CStr(Convert.ToInt32("  7 ")), "7")
    End Sub
End Module

File: tests/csharp/csharp_conversion_methods/convert_to_int32_from_string.cs
// vybe-test: csharp/csharp_conversion_methods/convert_to_int32_from_string
// origin: languages/csharp/tests/csharp/test_csharp_conversion_methods.rs

using static __Harness;

// conversion_methods
__P((System.Convert.ToInt32("99")).ToString());
__Check("99");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
